src/nativeMain/kotlin/eclairClient.kt

```diff
@@ -0,0 +1,179 @@
+import io.ktor.client.*
+import io.ktor.client.engine.cio.*
+import io.ktor.client.plugins.*
+import io.ktor.client.plugins.auth.*
+import io.ktor.client.plugins.auth.providers.*
+import io.ktor.client.request.*
+import io.ktor.client.statement.*
+import io.ktor.http.*
+import io.ktor.util.*
+
+
+interface IEclairClient {
+    suspend fun getInfo(password: String, host: String): Result<String>
+    suspend fun connectUri(password: String, host: String, uri: String): Result<String>
+    suspend fun connectManual(password: String, host: String, nodeId: String, manualHost: String): Result<String>
+    suspend fun connectNodeId(password: String, host: String, nodeId: String): Result<String>
+    suspend fun disconnect(password: String, host: String, nodeId: String): Result<String>
+}
+
+class EclairClient : IEclairClient {
+    override suspend fun getInfo(password: String, host: String): Result<String> {
+        val httpClient = HttpClient(CIO) {
+            install(Auth) {
+                basic {
+                    credentials {
+                        BasicAuthCredentials(username = "", password = password)
+                    }
+                }
+            }
+        }
+
+        return runCatching {
+            val response: HttpResponse = httpClient.post("$host/getinfo")
+            if (response.status == HttpStatusCode.Unauthorized) {
+                throw Exception("Unauthorized: Incorrect password")
+            } else {
+                response.bodyAsText()
+            }
+        }.onFailure { e ->
+            if (e is ClientRequestException && e.response.status == HttpStatusCode.Unauthorized) {
+                throw Exception("Unauthorized: Incorrect password")
+            }
+        }.also {
+            httpClient.close()
+        }
+    }
+
+    @OptIn(InternalAPI::class)
+    override suspend fun connectUri(password: String, host: String, uri: String): Result<String> {
+        val httpClient = HttpClient(CIO) {
+            install(Auth) {
+                basic {
+                    credentials {
+                        BasicAuthCredentials(username = "", password = password)
+                    }
+                }
+            }
+        }
+
+        return runCatching {
+            val response: HttpResponse = httpClient.post("$host/connect") {
+                parameter("uri", uri)
```

Comment:
I'm not sure but this might be the place where I'm not doing something right. 
I triedd to use this logic from [here](https://ktor.io/docs/request.html#url-fragment). I think what  I've written is completely wrong. 
But locally I tried using a method called [setBody](https://ktor.io/docs/request.html#body)
It looked like:
```
val response: HttpResponse = httpClient.post("$host/connect") {
                setBody(uri)
            }
```
But this also doesn't works and gives an error.

Reply:
Okay, I was finally able to figure it out after trying different permutations from [ktor requests docs](https://ktor.io/docs/request.html).
I'll update the PR, soon with the changes.